### PR TITLE
sci boxes

### PR DESCRIPTION
**Pull request by tomisme**
An experiment in connecting mini SCI environments

Let's say you could put executable code in Athens' blocks.

Some questions:
 - In what order do we evaluate our blocks?
 - How do we pass data in and out of our blocks?
 - How do we handle async code?

Attempted approach:
 - Blocks are passed the evaluated result of their parent (`*1`)

Some other approaches:
 - Blocks inherit the environment of their parent
 - Blocks mutate a global environment
 - Blocks are babashka pods?

Fun stuff to try:
 - Pass in the datascript connection
 - `spit`/`slurp` to IPFS etc.

### CODE REVIEW

**Comment by tomisme** [comment]
image

**Comment by tomisme** [comment]
Totally understand if folks aren't sold on devcards, or think experiments like this should be in a separate project / codebase :smile:

File: src/cljs/athens_devcards/db.cljs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(ns athens-devcards.db
```
Comment (jeroenvandijk) [inline src/cljs/athens_devcards/db.cljs:1]: I think we can do `athens.devcards.db`?

File: src/cljs/athens_devcards/sci_boxes.cljs [hunk 2]
```diff
@@ -0,0 +1,308 @@
+(ns athens-devcards.sci-boxes
```
Comment (jeroenvandijk) [inline src/cljs/athens_devcards/sci_boxes.cljs:1]: `athens-devcards.sci-boxes` -> `athens.devcards.sci-boxes`

File: src/cljs/athens_devcards/db.cljs (hunk 1 above)
Comment (tomisme) [reply src/cljs/athens_devcards/db.cljs:1]: Yeah sounds good :)

File: src/cljs/athens/devcards/sci_boxes.cljs [hunk 3]
```diff
@@ -0,0 +1,310 @@
+(ns athens.devcards.sci-boxes
+  (:require
+    [cljsjs.react]
+    [cljsjs.react.dom]
```
Comment (tomisme) [inline src/cljs/athens/devcards/sci_boxes.cljs:4]: I'm not sure why these react imports are necessary but I get "React is not defined" when they're not there

**Comment by tomisme** [comment]
> Really cool @tomisme! Given your questions in the PR, would you like to merge or are you still working on this?

Happy to merge, I won't be able to do any more work on it for a little while. That way others can play around with it :)

Should I squash my commits first?

**Comment by tangjeff0** [comment]
It can be done on GH @tomisme. Just share the final commit here and @jeroenvandijk or I will copy it in. :)

**Comment by tomisme** [comment]
> It can be done on GH @tomisme. Just share the final commit here and @jeroenvandijk or I will copy it in. :)
> 
> Or we can squash directly!

Excellent. Yeah I think that's the last of it, ready when you are :)

**Comment by tangjeff0** [comment]
I'll let @jeroenvandijk finish the PR because he reviewed already and he's been working with sci more closely. To respond to your comment @tomisme , though, the reason you need to require 
```
[cljsjs.react]
[cljsjs.react.dom]
```
is because React needs to be shimmed when using `shadow-cljs` @tomisme. See https://github.com/bhauman/devcards/issues/156

**Comment by tangjeff0** [comment]
`http://localhost:3001/cards.html` doesn't work for me. I have to open the file from my filesystem, e.g. `file:///athens/public/cards.html#!/athens.devcards.block`

Nothing even runs on `localhost:3001`, my build runs on `localhost:3000` still. Thoughts? @jeroenvandijk @tomisme

**Comment by tomisme** [comment]
Anything at localhost:3000/cards.html?

**Comment by avichalp** [comment]
For me, It is available on `localhost:3000`.  

@tangjeff0: You can also start a separate server using `lein devcards` then it will available on `3001` too.

**Comment by tangjeff0** [comment]
Works now @tomisme @avichalp ! Previously I was getting the default _All Pages_ view before at `localhost:3000/cards.html` but good now!

**Pull request merged**